code/modules/mob/hear_say.dm

```diff
@@ -166,13 +166,21 @@
 	track = handle_track(message, verb, speaker, speaker_name, follow_target, hard_to_hear)
 
 	if(!can_hear())
-		if(prob(20))
+		var/near_intercom = FALSE
+		for(var/obj/item/radio/intercom/intercom in view(5, src))
+			if(get_dist(src, intercom) <= intercom.canhear_range)
+				near_intercom = TRUE
+				break
+
+		if(near_intercom)
+			return
```

Comment:
being near an intercomm shouldnt prevent you from getting this message. Instead it should either
- only show up if you're wearing a headset (it would be nice if you could make a proc on /human for this)
- dont show up if the radio source is an intercomm (would need a large refactoring of /hear_radio, so i dont recommend it)